src/Host/Broker/Impl/Logging/FileLogger.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.R.Host.Broker.Logging {
+    internal sealed class FileLogger : ILogger, IDisposable {
+        private readonly string _category;
+        private volatile StreamWriter _writer;
+
+        public FileLogger(string category, StreamWriter writer) {
+            _category = category;
+            _writer = writer;
+        }
+
+        public void Dispose() {
+            _writer = null;
+        }
+
+        private sealed class Scope : IDisposable {
```

Comment:
Yes, for BeginScope. We can do something more useful here later if we want to - I haven't used scopes in our own logging so far, but that will likely change.

---
In reply to: [74831777](https://github.com/Microsoft/RTVS/pull/2173#discussion_r74831777) [](ancestors = 74831777)